plugin/trino-phoenix5/src/main/java/io/trino/plugin/phoenix5/PhoenixPageSource.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.phoenix5;
+
+import io.trino.spi.Page;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.RowBlock;
+import io.trino.spi.connector.ConnectorPageSource;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+public class PhoenixPageSource
```

Comment:
I would consider renaming this to something like `MergeRowIdPopulatingPageSource` or `PhoneixPageSourceWithMergeId` to clarify that it's only for adding merge row ids. Without merge, we don't use this class, so the current name is a bit too general.

Reply:
After updated using `ColumnAdaptation` I think the problem is fixed, we would always return `PhoenixPageSource`.